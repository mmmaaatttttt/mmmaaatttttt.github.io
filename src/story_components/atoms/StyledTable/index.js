import styled from "styled-components";
import PropTypes from "prop-types";

const StyledTable = styled.table`
  margin: ${props => props.margin};

  th,
  td {
    text-align: center;
    padding: ${props => props.padding};
  }
`;

StyledTable.propTypes = {
  fontSize: PropTypes.string.isRequired,
  margin: PropTypes.string.isRequired,
  padding: PropTypes.string.isRequired
};

StyledTable.defaultProps = {
  margin: "0 0 1.44rem 0",
  padding: "0.5rem 0"
}

export default StyledTable;
