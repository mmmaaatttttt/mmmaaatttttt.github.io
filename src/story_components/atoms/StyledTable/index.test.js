import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import StyledTable from ".";

it("renders successfully", () => {
  shallow(<StyledTable />);
});

it("matches snapshot with props", () => {
  const wrapper = shallow(
    <StyledTable margin="1rem">
      <thead>
        <tr>
          <th>1</th>
          <th>2</th>
        </tr>
        <tr>
          <td>3</td>
          <td>4</td>
        </tr>
      </thead>
    </StyledTable>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
