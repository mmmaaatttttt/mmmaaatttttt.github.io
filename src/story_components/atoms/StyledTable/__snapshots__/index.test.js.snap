// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  margin: ",
          [Function],
          ";

  th,
  td {
    text-align: center;
    padding: 0.5rem 0;
  }
",
        ],
      },
      "defaultProps": Object {
        "margin": "0 0 1.44rem 0",
      },
      "displayName": "styled.table",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "margin": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "table",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  margin="1rem"
>
  <thead>
    <tr>
      <th>
        1
      </th>
      <th>
        2
      </th>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        4
      </td>
    </tr>
  </thead>
</StyledComponent>
`;
